flow/__main__.py

```diff
@@ -89,12 +131,53 @@ def main():
         help="Specify a specific to template to use.",
     )
 
+    # the flow template command
+    parser_template = subparsers.add_parser(
+        "template", help="Create and manage custom user templates."
+    )
+    # Set a flag to indicate that flow template was called without a subparser.
+    parser_template.set_defaults(bare_template=True)
```

Comment:
I could see `template list` potentially. I am not sure how many more we would have, but in the case we do add more this does allow us to easily do so. I do get though that we are being somewhat hacky with using `set_defaults` in this way.

Reply:
Cool. Just wanted your thoughts on that, I also considered `list`. I am fine with the current implementation, setting a default variable is the same approach we use in other CLI code.

Reply:
I agree a `template list` command would be useful.

Reply:
I'm not sure why this default variable is necessary. Without it, don't you get get a (admittedly slightly less informative) error message? It should be the difference between
```
# What I think you get with your version
usage: flow template [arg list]

DESCRIPTION
```
and
```
# What you normally get, which is enough IMO
usage: flow template [arg list]
```
If I'm not wrong, I don't think the extra description is worth this hackery.

Reply:
@vyasr this is the base output which does not mention create at all,
`usage: __main__.py [-h] [--debug] [--version] {init,template} ...`. A user would have to call with `--help` for any information on using the command. I am fine with removing it just let me know what you prefer.

Reply:
I think that's the expected behavior for most command line tools on Unix, a complete help is only printed if requested otherwise you just get a basic usage message. It's certainly not universally followed, but it's the most common, e.g.
```
(main) it-HP-Z8-G4-Workstation% less
Missing filename ("less --help" for help)
```
However, what I think would be more in line with standard help while also being useful is giving some idea of subparser arguments:
```
(main) it-HP-Z8-G4-Workstation% flow template
usage: flow [-h] [--debug] [--version] {init,template} ...
```
What we should see here is basic help for the subparser. For example, here's what signac gives:
```
(main) it-HP-Z8-G4-Workstation% signac move
usage: signac move [-h] project job_id [job_id ...]
signac move: error: the following arguments are required: project, job_id
```
If you can modify the subparsers to provide that kind of information (in this case, the `create` argument for `flow template`), that would be fantastic.